src/App.tsx

```diff
@@ -15,47 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = string[];
+
+const SORT_FIELD_BY_LETTERS = 'letters';
+const SORT_FILED_BY_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods: Goods,
+  { sortField, reverseField }: { sortField: string, reverseField: boolean },
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_BY_LETTERS:
+          return a.localeCompare(b);
+        case SORT_FILED_BY_LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, reverseField },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SORT_FIELD_BY_LETTERS)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_BY_LETTERS,
+          })}
```

Comment:
If we use the classnames library, we should use it by dividing classes into different parts if possible.
```suggestion
          className={cn('button', 'is-info', {
            'is-light': sortField !== SORT_FIELD_BY_LETTERS,
          })}
```
Fix this in all code.